mantidproject/mantid
# Fix log splitting issue in FilterEvents

Description of work.

**To test:**

1. Find an EventNexus file to split from.
2. Test using TableWorkspace for splitters
  - Refer to the doc test of **FilterEvents** to create a TableWorkspace
  - Run *FilterEvents* with this TableWorkspace
3. Test splitting sample logs for splitters given by MatrixWorkspace and TableWorkspace
  - Check the sample logs of the output workspaces from step (2). 
  - Run **GenerateEventsFilter** to create MatrixWorkspace splitters by set 'FastLog'  to True.
  - Run **FilterEvents** with the MatrixWorkspace splitterd and check sample logs to be properly split
4. Check whether all the output workspaces has a sample log named *splitter*, which reflects the splitter that is applied to it.

Fixes #16756.
Fixes #12535.


*The change is reported in the FrameWork section in release note*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
